source/numeric-range.d.ts

```diff
@@ -0,0 +1,68 @@
+type Infinities = 'Infinity' | '-Infinity';
+
+/**
+A `number` that is an integer.
+
+Use-case: Validating and documenting parameters.
+
```

Comment:
Are there any open TS issues about adding an integer type that we could link to here?

Reply:
https://github.com/microsoft/TypeScript/issues/43505 seems to be the best fit